### PR TITLE
Remove and automate docs generation

### PR DESCRIPTION
**Pull request by Blacksmoke16**
* Adds a GH Actions workflow/job that builds/pushes generated docs to a `gh-pages` branch
  * Runs on every push to `master`
  * Also builds docs for the required `amq-protocol` dependency
    * Makes navigating docs a bit easier as types are now correctly linked

Example build: https://github.com/Blacksmoke16/amqp-client.cr/runs/1126200835?check_suite_focus=true
Example output: https://github.com/Blacksmoke16/amqp-client.cr/tree/gh-pages
Example docs: https://blacksmoke16.github.io/amqp-client.cr/

NOTE:  Currently Crystal docker images are a bit dated.  Could update the image/remove the need to manually install rsync once there are more proper images resulting from https://github.com/crystal-lang/distribution-scripts/pull/68.

### CODE REVIEW

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+      - name: Install Build Dependencies
+        run: apk add --update rsync
+      - uses: actions/checkout@v2
+      - name: Install Dependencies
+        run: shards install --production
+      - name: Build
+        run:  crystal docs lib/amq-protocol/src/amq-protocol.cr src/amqp-client.cr
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@3.5.1
```
Comment (dentarg) [inline .github/workflows/deployment.yml:22]: Can we use the latest version, 3.6.1? (https://github.com/JamesIves/github-pages-deploy-action/releases)

Comment (Blacksmoke16) [reply .github/workflows/deployment.yml:22]: Ah jeez, was on the 2nd page :see_no_evil:.

**Review by dentarg — APPROVED** [review]
👍 

(The action used for deploying to GitHub Pages is https://github.com/JamesIves/github-pages-deploy-action#readme if anyone wants to read more about it)

**Comment by dentarg** [comment]
@carlhoerberg WDYT?

**Comment by carlhoerberg** [comment]
I like that the docs are built for amq-protocol too. But can we get rid if the container and install crystal via snap instead? Less to download, less layers

**Comment by Blacksmoke16** [comment]
@carlhoerberg Probably, but would it be worth it?  The workflow already only takes 30s with the container being 9s of that.

**Comment by carlhoerberg** [comment]
Thanks! updated the website to use the gh-pages branch

**Pull request merged**